### DISCUSSION TITLE
Can you implement "Stay on Top" on Wayland Gnome?

### DISCUSSION THREAD

**Discussion by HubKing** [tomasklaen/uosc#510] [Q&A]
MPV's in-built "Stay on Top" does not work on Wayland Gnome. As far as I know, MPV developers do not want to deal with DE-specific API's or provide full UI, so currently, I have to press Alt+Space and click "Always on Top". This is inconvenient. The MPV front-ends I have tested had "stay on top" feature, but they added their own problems, so I have been using vanilla MPV mostly. But, of course, I am willing to switch to a better UI, if there is one.

I have tested this app, and Shift+T did not make it stay on top. Is it very difficult to implement on top of vanilla MPV? If not, is there any plan to implement that feature?

**Comment by christoph-heinrich** [comment]
Is there some way you can tell gnome that the window should be on top via the command line? Can be dbus or anything else.
If that exists, then you can create a keybinding that executes that command and even a control in uosc to have a button in the UI.
Otherwise I don't think there is anything we can do.

**Comment by hooke007** [comment]
> I have tested this app, and Shift+T did not make it stay on top

I think you totally misunderstand uosc. It is not frontend. It is the alternative script of built-in osc.lua.
Any vanilla MPV using uosc is still the vanilla MPV.

**Comment by ceed0** [comment]
> I have to press Alt+Space and click "Always on Top". This is inconvenient.

You can configure a separate keybinding for it like that
`gsettings set org.gnome.desktop.wm.keybindings always-on-top "['<Super>t']"`
